tests/src/python/test_qgspdfrenderer.py

```diff
@@ -0,0 +1,97 @@
+"""QGIS Unit tests for QgsPdfRenderer
+
+From build dir, run: ctest -R QgsPdfRenderer -V
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+
+import os
+import unittest
+
+from qgis.PyQt.QtCore import (
+    Qt,
+    QRectF
+)
+from qgis.PyQt.QtGui import (
+    QImage,
+    QPainter
+)
+from qgis.core import (
+    QgsPdfRenderer
+)
+from qgis.testing import start_app, QgisTestCase
+
+from utilities import unitTestDataPath
+
+start_app()
+
+TEST_DATA_DIR = unitTestDataPath()
+
+
+class TestQgsPdfRenderer(QgisTestCase):
+
+    @classmethod
+    def control_path_prefix(cls):
+        return "pdf"
+
+    def test_pdf_properties(self):
+        """Test PDF properties"""
+        pdf_path = os.path.join(TEST_DATA_DIR, 'sample_pdf.pdf')
+        renderer = QgsPdfRenderer(pdf_path)
+        self.assertEqual(renderer.pageCount(), 2)
+        self.assertEqual(renderer.pageMediaBox(-1), QRectF())
+        self.assertEqual(renderer.pageMediaBox(0),
+                         QRectF(0.0, 0.0, 842.0, 595.0))
+        self.assertEqual(renderer.pageMediaBox(1),
+                         QRectF(0.0, 0.0, 420.0, 595.0))
+        self.assertEqual(renderer.pageMediaBox(2), QRectF())
+
+    def test_pdf_render(self):
```

Comment:
maybe add a test by providing something that is not a PDF to QgsPdfRenderer ?

Reply:
Sure, done!